### DISCUSSION TITLE
Integration of tree-sitter grammar for the Smithy language

### DISCUSSION THREAD

**Discussion by Baccata** [community/community#36622] [Other Feature Feedback, Questions, & Ideas]
Hey folks, 

#### Regarding the Smithy language 

[Smithy](https://awslabs.github.io/smithy/) is the interface definition language (IDL)  that AWS has built to use as base for the multiple code-generators that power their SDKs. It is a very well designed IDL, in that it is protocol agnostic, and can be used to describe data types and interfaces in a lot of different contexts. It was open-sourced a few years ago, and is gaining a bit of traction. We (DisneyStreaming) use it as a mean to describe the APIs that form our platform, and are building bespoke tooling on top of it. 

Having syntax highlighting and other features like code navigation, would be amazing for productivity. 

A friend of mine has created a [tree-sitter grammar](https://github.com/indoorvivants/tree-sitter-smithy) for it, to get syntax highlighting in neovim. 

My questions are the following : 

1. could this grammar be integrated to power syntax highlighting / code navigation / code search in github ? 
2. is there a requirement as to where the grammar should live for it to be the case ? 
3. depending on the response to 2., what happens if/when the language evolves and the grammar needs to be updated ? 

NB : somebody from the awslabs organisation has opened a PR against linguist to reference the textmate grammar used in a vscode extension : https://github.com/github/linguist/pull/6110

#### Collateral questions (not Smithy related) 

More generally, it is an impediment for "new" languages to not get syntax highlighting in github from the get-go, and for the barrier of entry for getting it to be fuzzy, when it's quite obvious that a languages are bound to get more traction by having basic support for syntax highlighting in github. 

I understand the fact that allowing users to plug arbitrary third-party grammars to their repos poses a security risk, but the feature is so crucial for any language that I'm actually contemplating creating a browser extension that would allow prototype languages to get syntax highlighting in github. This leads me to the following question :

4. Before I dive in this rabbit hole, is the support for opt-in, third-party grammar on Github's roadmap ? Even in a textmate-based / front-end-handled fashion ? 

Kind regards, and thank you in advance for your answers !

**Comment by ossanna16** [comment]
Hey there @Baccata and thank you for asking a good question! I'm glad you found our community 🙂

**Comment by Baccata** [comment]
@ossanna16 I appreciate you appreciating my question 😄  is there anyone that could be nudged to provide an answer ?

**Comment by ossanna16** [comment]
Hi there @Baccata! Unfortunately I don't! I'm sorry!

**Comment by Baccata** [comment]
After a quick look, looks like the people who replied in this discussion might be able to answer the question : https://github.com/orgs/community/discussions/10934

@colinwm, @dcreager, apologies for the intrusive ping from a stranger (me), but you may be able to answer ? Thanks in advance
